test/AelinDeal.t.sol

```diff
@@ -2,17 +2,215 @@
 pragma solidity 0.8.6;
 
 import "ds-test/test.sol";
+import "forge-std/Test.sol";
 import {AelinDeal} from "../contracts/AelinDeal.sol";
+import {AelinPool} from "../contracts/AelinPool.sol";
```

Comment:
the new tests are great. I will go through them in detail before we merge but scrolled through them and it is really nice to see all the changes